vars/deploymentPipeline.groovy

```diff
@@ -42,8 +42,9 @@ private def parseParams(envs, svcs) {
 def call(p = [:]) {
     envs = p['environments']
     svcs = p['services']
+    params = p['params'] ?: []
```

Comment:
One minor thing here, can we call this 'extraParams' just to make its purpose a little more clear? :) Otherwise, looks good.

I have a feeling this name may also be the cause of your test failure. The 'params' name is actually a reserved name in Jenkinsfiles -- it is used to access the current job parameters.